src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +22,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    public boolean equals(Rectangle rectangle) {
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle compareRectangle = (Rectangle) rectangle;
+            return ((width == null && compareRectangle.getWidth() == null)
```

Comment:
```suggestion
            return (width == compareRectangle.getWidth()
```
this can be replaced